lua/nvim-treesitter-refactor/highlight_definitions.lua

```diff
@@ -9,13 +10,19 @@ local cmd = api.nvim_command
 local M = {}
 
 local usage_namespace = api.nvim_create_namespace('nvim-treesitter-usages')
+local last_node = nil
 
 function M.highlight_usages(bufnr)
-  M.clear_usage_highlights(bufnr)
-
   local node_at_point = ts_utils.get_node_at_cursor()
-  local references = locals.get_references(bufnr)
+  -- Don't calculate usages again if we are on the same node.
+  if node_at_point == last_node and M.has_highlights(bufnr) then
+    return
+  else
+    last_node = node_at_point
+  end
 
+  M.clear_usage_highlights(bufnr)
+  local references = locals.get_references(bufnr)
   if not node_at_point or not vim.tbl_contains(references, node_at_point) then
```

Comment:
`if not node_at_point` could be a short circuit, in that case we have not to `locals.get_references()`, and should we clear_usage_highlights also? especially with `clear_on_cursor_move = true`

Reply:
additionally, `locals.get_references()` here and the following `locals.find_definition()` can be merged? at least, `references` could be reused if possible.

Reply:
Looks like get_references is just a check to avoid doing another call, but in the other call we loop through all references again, I think we can remove it :+1: